### PR TITLE
Feat exec wasm with json file using single ecall for all wasms inside enclave

### PR DESCRIPTION
**Pull request by Astewart1510**
### Single ecall for all wasm

Updated PR

This PR trims down the repo. It combines all wasm execution ecalls into a single ecall that can run for multiple wasms depending on the wasm destination. All existing wasms - mean, median and stdev - have been updated to first standardise the data to floats and then executing the logic. 
Therefore all numerical types  - integers and floats - mixed into a the same array or json data column can be computed over.

##UPDATE

updated entire repo
- created test_data directory
- updated ecalls to consume serialised data along with schema
- imported a result buffer created by the enclave into wasm binaries
- updated get-mean wasm binary to deserialise data based on a schema presented to the wasm binary
- enclave can access result buffer and print to terminal (in SW print is allowed only)
- All schema must follow the standard online json schema found here - http://json-schema.org/draft-07/schema#. 

### To Test
- make all
- cd bin
- ./app

### CODE REVIEW

**Comment by Astewart1510** [comment]
@billguo99 check it out and merge.. pretty sure its all good :) but I updated quite a bit of code and trimmed a lot of it... one ecall for all wasm exeuctions.. and the wasms can execute over floats and integers at the same time in the same array. All results are returned as floats.

**Comment by billguo99** [comment]
I ran `./prepare-1.1.6-intel-2.20.sh` from the `rust-sgx-sdk-dev-env` repo to setup my environment.

I'm getting the following error, after `make all`, `cd bin`, `./app`:
```
~/workspace/wasm-exec-sgx/bin feat-exec-wasm-with-json-file ❯ ./app                                                                                                    11:54:26
[+] Init Enclave Successful 178095113895938!
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 13, kind: PermissionDenied, message: "Permission denied" }', src/main.rs:73:38
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

File: app/src/main.rs [hunk 1]
```diff
@@ -18,71 +18,34 @@
...
+extern crate serde_json;
+use std::path::Path;
+use std::{
+    fs::{self, File},
+    io::Read,
+};
 
-static WASM_FILE_MEAN_INT: &str = "get_mean_int.wasm";
-static WASM_FILE_MEAN_FLOAT: &str = "get_mean_float.wasm";
+static WASM_FILE_MEDIAN: &str = "get_median_int.wasm";
```
Comment (billguo99) [inline app/src/main.rs:32]: I think it should be "get_median_wasm.wasm"

File: app/src/main.rs [hunk 2]
```diff
@@ -118,71 +100,67 @@ fn main() {
...
+    )
+    .unwrap();
+    let serialized_data_median: Vec<u8> = serde_json::to_vec(&data_median).unwrap(); // Create a new byte array that holds the serialized JSON data
+    let binary_median = fs::read(WASM_FILE_MEDIAN).unwrap();
+    let mut result_out_median = 0f32;
+
+    //// Mean test
+    // Numbers float and integer mixed
+    let data_mean = read_data_from_json(
+        "/root/workspace/wasm-exec-sgx/get-mean-wasm/test_data.json",
```
Comment (billguo99) [inline app/src/main.rs:117]: The file path needs to change, "/home/bill/workspace..." worked for me for my system.

File: app/src/main.rs [hunk 3]
```diff
@@ -118,71 +100,67 @@ fn main() {
...
+    )
+    .unwrap();
+    let serialized_data_mean: Vec<u8> = serde_json::to_vec(&data_mean).unwrap(); // Create a new byte array that holds the serialized JSON data
+    let binary_mean = fs::read(WASM_FILE_MEAN).unwrap();
+    let mut result_out_mean = 0f32;
+
+    // Sd test
+    // Numbers float and integer mixed
+    let data_sd = read_data_from_json(
+        "/root/workspace/wasm-exec-sgx/get-sd-wasm/test_data.json",
```
Comment (billguo99) [inline app/src/main.rs:128]: The file path needs to change, "/home/bill/workspace..." worked for me for my system.

File: app/src/main.rs [hunk 4]
```diff
@@ -118,71 +100,67 @@ fn main() {
...
-
-    let mut result_out_mean_int = 0i32;
-    let mut result_out_mean_float = 0f32;
-
-    let mut result_out_sd_int = 0f32;
-    let mut result_out_sd_float = 0f32;
+    //// Mean test
+    // Numbers float and integer mixed
+     let data_median = read_data_from_json(
+        "/root/workspace/wasm-exec-sgx/get-median-wasm/test_data.json",
```
Comment (billguo99) [inline app/src/main.rs:106]: The file path needs to change, "/home/bill/workspace..." worked for me for my system.

**Pull request merged**